pkg/apply/event/event.go

```diff
@@ -75,92 +81,73 @@ type ErrorEvent struct {
 	Err error
 }
 
-//go:generate stringer -type=ApplyEventType
-type ApplyEventType int
+//go:generate stringer -type=ActionGroupEventType
+type ActionGroupEventType int
 
 const (
-	ApplyEventResourceUpdate ApplyEventType = iota
-	ApplyEventCompleted
+	Started ActionGroupEventType = iota
+	Finished
 )
 
+type ActionGroupEvent struct {
+	GroupName string
+	Action    ResourceAction
+	Type      ActionGroupEventType
+}
+
 //go:generate stringer -type=ApplyEventOperation
 type ApplyEventOperation int
 
 const (
-	ServersideApplied ApplyEventOperation = iota
+	NoneApply ApplyEventOperation = iota
+	ServersideApplied
 	Created
 	Unchanged
 	Configured
-	Failed
 )
 
 type ApplyEvent struct {
-	Type       ApplyEventType
-	Operation  ApplyEventOperation
-	Object     *unstructured.Unstructured
 	Identifier object.ObjMetadata
+	Operation  ApplyEventOperation
+	Resource   *unstructured.Unstructured
 	Error      error
 }
 
-//go:generate stringer -type=StatusEventType
-type StatusEventType int
-
-const (
-	StatusEventResourceUpdate StatusEventType = iota
-	StatusEventCompleted
-)
-
 type StatusEvent struct {
-	Type     StatusEventType
-	Resource *pollevent.ResourceStatus
+	Identifier       object.ObjMetadata
+	PollResourceInfo *pollevent.ResourceStatus
+	Resource         *unstructured.Unstructured
+	Error            error
 }
 
-//go:generate stringer -type=PruneEventType
-type PruneEventType int
-
-const (
-	PruneEventResourceUpdate PruneEventType = iota
-	PruneEventCompleted
-	PruneEventFailed
-)
-
 //go:generate stringer -type=PruneEventOperation
 type PruneEventOperation int
 
 const (
-	Pruned PruneEventOperation = iota
+	NonePrune PruneEventOperation = iota
```

Comment:
similar to `NonApply`, how about changing `NonePrune` to `PruneUnspecified` or `Unspecified`.

Reply:
Unfortunately I can't use the same name (like `Unspecified`) for the the different types. So they need to have a name that includes Apply/Prune/Destroy. I have renamed them to `(Apply|Prune|Destroy)Unspecified`